src/bson.c

```diff
@@ -1364,19 +1366,6 @@ void phongo_zval_to_bson(zval *data, php_phongo_bson_flags_t flags, bson_t *bson
 			return;
 	}
 
-	if (!ht_data || ZEND_HASH_GET_APPLY_COUNT(ht_data) > 1) {
-#if PHP_VERSION_ID >= 70000
-		if (Z_TYPE_P(data) == IS_OBJECT && instanceof_function(Z_OBJCE_P(data), php_phongo_serializable_ce TSRMLS_CC)) {
-#endif
-			if (!Z_ISUNDEF(obj_data)) {
-				zval_ptr_dtor(&obj_data);
-			}
-#if PHP_VERSION_ID >= 70000
-		}
-#endif
-		return;
-	}
-
```

Comment:
This block was only necessary for cleaning up after the exception when `bsonSerialize()` does not return an array or stdClass. I moved the `if (!Z_ISUNDEF(obj_data)) { zval_ptr_dtor(&obj_data); }` condition up to immediately follow that exception and changed the original `break` to a `return`.

I don't believe the Serializable `instanceof` for PHP 7 was especially relevant, as `zval *obj_data` is initialized to `NULL` on PHP 5.x and only ever assigned when we're dealing with a Serializable (it stores the return value of `bsonSerialize()`. AFAICT, this was dead code.